# Importing libraries

In [1]:
import pandas as pd
import glob

# Read all the csv files created by scraping and structed data and create combined data frame

In [2]:
path = 'C:/Users/mrash/Desktop/ISB/DCCP/Group Project'
csv_files_list = glob.glob(path + "/*.csv")
df_list = (pd.read_csv(file) for file in csv_files_list)

combined_data   = pd.concat(df_list, ignore_index=True)

In [3]:
combined_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13561 entries, 0 to 13560
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Unnamed: 0     13561 non-null  int64 
 1   Species_Name   13561 non-null  object
 2   Family_name    13534 non-null  object
 3   Local_name     712 non-null    object
 4   Common_name    3635 non-null   object
 5   Parts_used     6816 non-null   object
 6   Medicinal_Use  13374 non-null  object
dtypes: int64(1), object(6)
memory usage: 741.7+ KB


# remove indexed column (Unnamed: 0)

In [4]:
combined_data.drop(['Unnamed: 0'], axis=1, inplace=True)

# data clean-up

In [5]:
combined_data.fillna('', inplace=True)
combined_data["Medicinal_Use"].replace("None knownh3", "", regex=True , inplace=True)
combined_data["Medicinal_Use"].replace("None known", "", regex=True, inplace=True)
combined_data.drop_duplicates(keep='first' , inplace= True)
combined_data.drop_duplicates(subset=["Species_Name", "Family_name"], keep='first',  inplace= True)

# create a csv file for combine data after cleaning

In [6]:
combined_data.to_csv('data_plant_combined_all.csv')

# Create json file for combine data after cleaning

In [7]:
combined_data.to_json('data_plant_combined.json',orient ='records')

In [8]:
combined_data

Species_Name     Family_name Local_name  \
0          Abarema cochliacarpos        Fabaceae              
1                Abarema jupunba        Fabaceae              
2          Abelmoschus angulosus       Malvaceae              
3            Abelmoschus caillei       Malvaceae              
4           Abelmoschus crinitus       Malvaceae              
...                          ...             ...        ...   
13554   Trigonelia Foenu-graceum     Leguminosae              
13555           Vernonia biafrae      Compositae              
13558       Wood fordia uniflora      Lythraceae              
13559       Xeromorphus nilotica       Rubiaceae              
13560  Xysmalobium heudelotiamum  Asclopiadaceae              

                      Common_name                        Parts_used  \
0                      Barbatimão                         bark,wood   
1                                  root,leave,bark,whole plant,seed   
2                                                                     
3               West African Okra                   root,leave,seed   
4                                                             leave   
...                           ...                               ...   
13554                   Fenugreek                  Whole plant seed   
13555                                                  Leaves, root   
13558                                                 Dried flowers   
13559  Busky gardenia, emetic nut                      Fruits, root   
13560                                                         Roots   

                                           Medicinal_Use  
0      The wood is powdered and used in domestic medi...  
1      The whole plant is said to be a parasiticide b...  
2                                                         
3      This plant is very closely related to okra Abe...  
4      The leaves are used in the treatment of dysent...  
...                                                  ...  
13554  Swellings, leucorrhoea, astringent, throat inf...  
13555                             Malaria, boils, wounds  
13558            Astringent, menorrhagia, ulcers, wounds  
13559  Emetic, expectorant, antispasmodic, dysentery,...  
13560                 Emetic, stomachache, bitter tonic.  

[13394 rows x 6 columns]

# EDA - Plant dataset

In [9]:
combined_data.shape

(13394, 6)

In [10]:
combined_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13394 entries, 0 to 13560
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Species_Name   13394 non-null  object
 1   Family_name    13394 non-null  object
 2   Local_name     13394 non-null  object
 3   Common_name    13394 non-null  object
 4   Parts_used     13394 non-null  object
 5   Medicinal_Use  13394 non-null  object
dtypes: object(6)
memory usage: 1.2+ MB


In [11]:
for col in combined_data.columns:
    print(col,combined_data[col].nunique())

Species_Name 13278
Family_name 361
Local_name 573
Common_name 3307
Parts_used 1265
Medicinal_Use 6844


In [12]:
combined_data[combined_data['Medicinal_Use']!='']

Species_Name     Family_name Local_name  \
0          Abarema cochliacarpos        Fabaceae              
1                Abarema jupunba        Fabaceae              
3            Abelmoschus caillei       Malvaceae              
4           Abelmoschus crinitus       Malvaceae              
5         Abelmoschus esculentus       Malvaceae              
...                          ...             ...        ...   
13554   Trigonelia Foenu-graceum     Leguminosae              
13555           Vernonia biafrae      Compositae              
13558       Wood fordia uniflora      Lythraceae              
13559       Xeromorphus nilotica       Rubiaceae              
13560  Xysmalobium heudelotiamum  Asclopiadaceae              

                      Common_name                        Parts_used  \
0                      Barbatimão                         bark,wood   
1                                  root,leave,bark,whole plant,seed   
3               West African Okra                   root,leave,seed   
4                                                             leave   
5                            Okra      root,leave,fruit,flower,seed   
...                           ...                               ...   
13554                   Fenugreek                  Whole plant seed   
13555                                                  Leaves, root   
13558                                                 Dried flowers   
13559  Busky gardenia, emetic nut                      Fruits, root   
13560                                                         Roots   

                                           Medicinal_Use  
0      The wood is powdered and used in domestic medi...  
1      The whole plant is said to be a parasiticide b...  
3      This plant is very closely related to okra Abe...  
4      The leaves are used in the treatment of dysent...  
5      The roots are very rich in mucilage having a s...  
...                                                  ...  
13554  Swellings, leucorrhoea, astringent, throat inf...  
13555                             Malaria, boils, wounds  
13558            Astringent, menorrhagia, ulcers, wounds  
13559  Emetic, expectorant, antispasmodic, dysentery,...  
13560                 Emetic, stomachache, bitter tonic.  

[7386 rows x 6 columns]

In [13]:
combined_data[combined_data['Medicinal_Use']=='']

Species_Name    Family_name Local_name      Common_name  \
2      Abelmoschus angulosus      Malvaceae                               
10       Abies guatemalensis       Pinaceae                               
11         Abobra tenuifolia  Cucurbitaceae             Cranberry Gourd   
14        Abrahamia louvelii  Anacardiaceae                               
16     Abrahamia thouvenotii  Anacardiaceae                               
...                      ...            ...        ...              ...   
12716      Zombia antillarum      Arecaceae                               
12722     Zornia leptophylla       Fabaceae                               
12726         Zygia racemosa       Fabaceae                  Marblewood   
13449       Bryoria africana  Cucurbitaceae                       Roots   
13450     Bursoma abyssinica  Melianthaceae                   Stem-bark   

                     Parts_used Medicinal_Use  
2                                              
10                                             
11                                             
14                                             
16                                             
...                         ...           ...  
12716                                          
12722                                          
12726                                          
13449                 Purgative                
13450  Anthelmintic, cathartic.                

[6008 rows x 6 columns]

# EDA Summary

1. there are total 13278 distinct Species
2. there are 361 family
3. there are 7386 medicinal plants
4. there are 6008 non medicinal plants